src/components/WordSearch/WordSearch.js

```diff
@@ -0,0 +1,250 @@
+import { QuestionContainer, Rectangle, Board, QuestionTypeText, HowToPlayText, WordBank, Word } from './WordSearchElements';
+import Letter from './Letter';
+import SuccessPopup from '../SuccessPopup/SuccessPopup';
+import axios from 'axios';
+
+function WordSearch(props) {
+
+    const [wordsFound, updateWordsFound] = useState([]);
+    const [wordSelected, setWordSelected] = useState("");
+    const [letterSelectedCoordinate, setLetterSelectedCoordinate] = useState({rowIndex: -1, colIndex: -1});
+    const [direction, setDirection] = useState("");
+    const [lettersInWordSelected, updateLettersInWordSelected] = useState([]);
+    const [lettersDisabled, updateLettersDisabled] = useState([]);
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        // Word has been found
+        if (props.wordBank.includes(wordSelected)) {
+            let indexOfWord = props.wordBank.indexOf(wordSelected);
+            updateWordsFound(wordsFound => wordsFound.concat(indexOfWord));
+            setWordSelected("");
+            setLetterSelectedCoordinate({rowIndex: -1, colIndex: -1});
+            setDirection("");
+            updateLettersDisabled(lettersDisabled => lettersDisabled.concat(lettersInWordSelected));
+            updateLettersInWordSelected([]);
+            console.log(wordsFound);
+        }
+
+        if (wordsFound.length === props.wordBank.length) {
+            wordSearchCompleted();
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [wordSelected])
+
+    const headers = {
+        'Content-type': 'application/json',
+        'Authorization': localStorage.getItem('access_token')
+    }
+
+    const wordSearchCompleted = () => {
+        const answer = {
+            answers: props.wordBank
+        }
+        axios.post(('http://localhost:8080/api/v0/questions?planet='+props.planet+'&questionNumber='+props.questionNumber), answer, {
+            headers: headers
+        }).then((res) => {
+            console.log(res);
+        }).catch(err => {
+            console.log(err);
+        });
+    }
+
+    return (
+        <BackgroundImg img={img}>
+            <IconButton onClick={() => {navigate("/")}}>
+                <HomeIcon style={{color: 'white', fontSize: '3.459vw', paddingLeft: '1%', paddingTop: '0.5%'}} />
+            </IconButton>
+            {wordsFound.length === props.wordBank.length && <SuccessPopup questionNumber={4} starsGained={3} redirect={"/"} />}
+            <QuestionContainer>
+                <QuestionTypeText>Question {props.questionNumber}: Word Search</QuestionTypeText>
+                <HowToPlayText>Find all of the words hidden in the grid! Click on the letters to form a word.</HowToPlayText>
+                <Line width={"100%"}/>
+                <Rectangle>
+                    <Board>
+                        <tbody>
+                            {props.boardLetters.map((items, rowIndex) => {
+                                return (
```

Comment:
Can this be externalized? It makes the html difficult to read

Reply:
It can be but, even if we make a separate component for it, it wont help with the readability too much since that component will contain all of the logic & be complicated as well.